docs/HyperIndex/migration-guide-v1-v2.md

```diff
@@ -0,0 +1,248 @@
+---
+id: migration-guide-v1-v2
+title: Migration Guide v1 to v2
+sidebar_label: Migration Guide v1 to v2
+slug: /migration-guide-v1-v2
+---
+
+NOTE: v2 is currently in rc phase (release candidate) and is not yet stable. Please refer to the [v1 documentation](https://docs.envio.dev/docs/HyperIndex) for the stable version.
+
+# Introduction
+
+V2 of HyperIndex is about streamlining the process of starting an indexer and optimizing it as you go. There are two big changes:
+
+- Handlers are now asynchronous, and by default 'loaders' aren't required at all - there is no need for the [async-mode](/docs/HyperIndex/async-mode).
+- Loaders (when used) are more expressive and connected via the return type to the context of the handler.
+  - In v1, you needed to use linked entities to load entity fields of other entities. This was unintuitive.
+    - In v2, you can directly access the fields of the loader the exact same way as you do in the handler, with an async 'get' function.
+  - In v1, you needed to call 'load' in the loader, and 'get' in the handler separately (or use labelled fields).
+    - In v2, you can use the return type of the loader to directly access the fields in the handler via the context, or you can call 'get' again.
+
+## Changes to Make
+
+### Handlers
+
+- Handlers are now asynchronous (you can add an `async` keyword before it).
+- You can use `handlerWithLoader` if you need a loader, otherwise use `handler` directly.
+- The 'get' function is now asyncronous, so add an `await` before those functions.
+- No labelled entities.
+
+### Loaders
+
+- Loaders are merged into the handlers using `handlerWithLoader`.
+- Loading linked entities is done directly with promices in the loader.
+- Loaders are completely optional - only use the if you care about high throughput indexing.
+- Loaders return the required entities which are then used in the handler.
+- `contractRegister` is used for dynamic contract registration. You don't add contracts to the indexer dynamically in the loader anymore.
+- The return type of the loader is used directly in the handler to access the loaded data. No need to re-'get' it again in the handler.
+
+### Configuration
+
+- There is no async-mode anymore, so you can remove `isAsync: true` from each of the events in your `config.yaml`.
+
+```diff
+- isAsync: true
+```
+
+- No entity labels or required entities.
+
+### Miscellaneous breaking changes and deprecations
+
+- For rescript, you need to use the built in bigint type instead of the `Ethers.BigInt` type.
+- Upgrade to rescript 11 uncurried mode (and use of newer rescript compiler).
+- The `context.Entity.load` function is deprecated and should be replaced with direct calls to `context.Entity.get` in the loader.
+- The `context.ParentEntity.loadField` functions are deprecated and should be replaced with direct calls to `context.ChildEntity.get`.
+<!-- TODO: lots more to put here -->
```

Comment:
Removed `Contract` and `Entity` suffixes among the ones I remember

Reply:
```suggestion
- Remove the `Contract` and `Entity` suffixes from generated code.
<!-- TODO: lots more to put here -->
```